src/Symfony/Component/HtmlSanitizer/Visitor/DomVisitor.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HtmlSanitizer\Visitor;
+
+use Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig;
+use Symfony\Component\HtmlSanitizer\Visitor\Model\Cursor;
+use Symfony\Component\HtmlSanitizer\Visitor\Node\BlockedNode;
+use Symfony\Component\HtmlSanitizer\Visitor\Node\DocumentNode;
+use Symfony\Component\HtmlSanitizer\Visitor\Node\Node;
+use Symfony\Component\HtmlSanitizer\Visitor\Node\TextNode;
+use function Symfony\Component\String\u;
+
+/**
+ * The DomVisitor iterate over the parsed DOM tree and visit nodes using the NodeVisitor.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class DomVisitor
+{
+    private array $allowedNodes;
+    private HtmlSanitizerConfig $config;
+
+    /** @var AttributeSanitizer\AttributeSanitizerInterface[][][] */
```

Comment:
Why the aliased namespace?

Reply:
Not an alias, but a sub-namespace of the current.

Reply:
Indeed, that's what I meant